scene/3d/ray_cast_3d.cpp

```diff
@@ -146,6 +147,12 @@ bool RayCast3D::get_exclude_parent_body() const {
 void RayCast3D::_notification(int p_what) {
 	switch (p_what) {
 		case NOTIFICATION_ENTER_TREE: {
+			if (debug_shape_color == Color(0.5, 0.7, 1.0)) {
+				debug_shape_color = get_tree()->get_debug_collisions_color();
```

Comment:
Colors use floating-point components, so use approximate equality to account for floating-point precision errors:

```suggestion
			if (debug_shape_color.is_equal_approx(Color(0.5, 0.7, 1.0))) {
				// Use the default debug shape color defined in the Project Settings.
				debug_shape_color = get_tree()->get_debug_collisions_color();
```